libnopegl/src/node_hblur.c

```diff
@@ -0,0 +1,677 @@
+    s->image = &src_priv->image;
+    s->image_rev = SIZE_MAX;
+
+    /* Disable direct rendering */
+    src_priv->supported_image_layouts = 1U << NGLI_IMAGE_LAYOUT_DEFAULT;
+
+    /* Override texture params */
+    src_priv->params.min_filter = NGLI_FILTER_LINEAR;
+    src_priv->params.mag_filter = NGLI_FILTER_LINEAR;
+    src_priv->params.mipmap_filter = NGLI_MIPMAP_FILTER_LINEAR;
+
+    s->map_image = &s->dummy_map_image;
+    s->map_rev = SIZE_MAX;
+    if (o->map) {
+        struct texture_priv *map_priv = (struct texture_priv *)o->map->priv_data;
+
+        /* Disable direct rendering */
+        map_priv->supported_image_layouts = 1U << NGLI_IMAGE_LAYOUT_DEFAULT;
+
+        /* Override texture params */
+        map_priv->params.min_filter = NGLI_FILTER_LINEAR;
+        map_priv->params.mag_filter = NGLI_FILTER_LINEAR;
+        s->map_image = &map_priv->image;
+    }
+
+    s->prefered_format = get_prefered_format(ctx->gpu_ctx);
+
+    struct texture_priv *dst_priv = o->destination->priv_data;
+    dst_priv->params.usage |= NGLI_TEXTURE_USAGE_COLOR_ATTACHMENT_BIT;
+
+    s->dst_is_resizeable = (dst_priv->params.width == 0 && dst_priv->params.height == 0);
+    s->pass2.layout.colors[0].format = dst_priv->params.format;
+    s->pass2.layout.nb_colors = 1;
+
+    const struct gpu_block_field block_fields[] = {
+        NGLI_GPU_BLOCK_FIELD(struct blur_params_block, radius, NGLI_TYPE_I32, 0),
+        NGLI_GPU_BLOCK_FIELD(struct blur_params_block, nb_samples, NGLI_TYPE_I32, 0),
+    };
+
+    const struct gpu_block_params block_params = {
+        .fields    = block_fields,
+        .nb_fields = NGLI_ARRAY_NB(block_fields),
+    };
+
+    int ret = ngli_gpu_block_init(gpu_ctx, &s->blur_params_block, &block_params);
+    if (ret < 0)
+        return ret;
+
+    if ((ret = setup_dummy_map(node)) < 0 ||
+        (ret = setup_pass1_pipeline(node)) < 0 ||
+        (ret = setup_pass2_pipeline(node)) < 0)
+        return ret;
+
+    return 0;
+}
+
+static int resize(struct ngl_node *node)
+{
+    int ret = 0;
+    struct ngl_ctx *ctx = node->ctx;
+    struct hblur_priv *s = node->priv_data;
+    struct hblur_opts *o = node->opts;
+
+    ngli_node_draw(o->source);
+    if (o->map)
+        ngli_node_draw(o->map);
+
+    struct texture_priv *src_priv = o->source->priv_data;
+    const int32_t width = src_priv->image.params.width;
+    const int32_t height = src_priv->image.params.height;
+    if (s->width == width && s->height == height)
+        return 0;
+
+    struct texture *dst = NULL;
+    struct texture *tex0 = ngli_texture_create(ctx->gpu_ctx);
+    struct texture *tex1 = ngli_texture_create(ctx->gpu_ctx);
+    struct rtt_ctx *pass1_rtt_ctx = ngli_rtt_create(ctx);
+    struct rtt_ctx *pass2_rtt_ctx = ngli_rtt_create(ctx);
+    if (!tex0 || !tex1 || !pass1_rtt_ctx || !pass2_rtt_ctx) {
+        ret = NGL_ERROR_MEMORY;
+        goto fail;
+    }
+
+    struct texture_params texture_params = (struct texture_params) {
```

Comment:
`(struct texture_params)` is unnecessary

Reply:
Removed.